### PR TITLE
HAI-2154 Add client for Profiili GraphQL API

### PR DESCRIPTION
**Pull request by corvidian**
# Description

Add a client for getting the verified name of the user from Profiili GraphQL API. This allows us to get the verified, official name of the user.

All the GraphQL libraries I found were really complicated with separate Gradle plugins, code generation and that fun stuff. We need just one static query, so using code generation to accomplish that felt like an overkill. Instead, add the GraphQL query in a resource file as a literal query and send it to the API server wrapped in an almost static JSON object.

Also, add a test data controller endpoint for querying the caller's own verified name. This was added to facilitate testing the endpoint in different environments, but also because that information is not readily available elsewhere.

## Type of change

- [ ] Bug fix 
- [X] New feature 
- [ ] Other

# Instructions for testing
Call the new test data controller from Swagger-UI. Make sure you have Suomi.fi login enabled.

# Checklist:

- [X] I have written new tests (if applicable)
- [X] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location:

### CODE REVIEW

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliClientITest.kt [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+                hasClass(VerifiedNameNotFound::class)
+                messageContains("User not authenticated")
+            }
+            assertThat(mockApiTokensApi.requestCount).isEqualTo(0)
+            verify { securityContext.accessToken() }
+        }
+
+        @Test
+        fun `throws exception when API tokens not found`() {
+            clearMocks(securityContext)
```
Comment (umeetiusbaar) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliClientITest.kt:85]: Is `clearMocks` needed here when there is `clearAllMocks` in `setUp`?

File: services/hanke-service/src/test/resources/application-test.yml [hunk 2]
```diff
@@ -13,6 +13,9 @@ haitaton:
   features:
     hanke-editing: true
     user-management: true
+  profiili-api:
```
Comment (umeetiusbaar) [inline services/hanke-service/src/test/resources/application-test.yml:16]: Where are these used?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliData.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package fi.hel.haitaton.hanke.profiili
+
+data class ProfiiliResponse(val data: ProfiiliData)
+
+data class ProfiiliData(val myProfile: MyProfile?)
+
+data class MyProfile(val verifiedPersonalInformation: Names?)
+
+data class Names(val firstName: String, val lastName: String, val givenName: String)
```
Comment (pitkni) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliData.kt:9]: The difference of firstName and givenName could be explained in a kdoc or something.

File: services/hanke-service/src/test/resources/application-test.yml (hunk 2 above)
Comment (corvidian) [reply services/hanke-service/src/test/resources/application-test.yml:16]: They are loaded into ProfiiliProperties.

Comment (umeetiusbaar) [reply services/hanke-service/src/test/resources/application-test.yml:16]: I meant the values `http://localhost:14678/*`. What service is this? And why these properties are under `test` when there is no other changes there? All other test changes are under `integrationTest`.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/PublicHankeController.kt [hunk 4]
```diff
@@ -40,5 +40,7 @@ class PublicHankeController(private val hankeService: HankeService) {
                 )
             ]
     )
-    fun getAll(): List<PublicHanke> = hankeService.loadPublicHanke().map { hankeToPublic(it) }
+    fun getAll(): List<PublicHanke> {
```
Comment (pitkni) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/PublicHankeController.kt:43]: Why the change? It's the same thing but with three lines.

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliClientITest.kt (hunk 1 above)
Comment (corvidian) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliClientITest.kt]: These were left over from some debugging I did earlier. Removed now.

File: services/hanke-service/src/test/resources/application-test.yml (hunk 2 above)
Comment (corvidian) [reply services/hanke-service/src/test/resources/application-test.yml:16]: I was planning on using that port for the mockWebService in ProfiiliClientITest, but I used two random ports instead. These could probably be replaced with tokens that are clearly non-functional. There should be something here to override the default value in application.yml, so that we don't accidentally connect to the actual backend from integration tests.

Both tests and integration tests use the test Spring profile, so they use the same application-test.yml. Integration tests can see this file from the src/test/ directory, but tests can't access anything in src/intergationTest/, so the file needs to be here under src/test/.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliData.kt (hunk 3 above)
Comment (corvidian) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliData.kt]: I added documentation here.

**Review by umeetiusbaar — APPROVED** [review]

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/PublicHankeController.kt (hunk 4 above)
Comment (corvidian) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/PublicHankeController.kt]: I used the endpoint for some early prototyping and didn't notice I left a change. Reverted this. Thank you for noticing.

**Review by pitkni — APPROVED** [review]
